### ISSUE TITLE
Getting a logError: TLError: unknown_read_err.

### ISSUE THREAD

**Issue by edgarnegronrabell** [NUKnightLab/TimelineJS3#693]
I'm trying to use your library, but I get this error by enabling debug as an option for the Timeline.
I've been setting breakpoints to try to figure out where the code isn't working and why.

logError: TLError: unknown_read_err
timeline.js:formatted:5559 TLError: unknown_read_err
    at error (https://cdn.knightlab.com/libs/timeline3/latest/js/timeline.js:11:145014)
    at d (https://cdn.knightlab.com/libs/timeline3/latest/js/timeline.js:11:113928)
    at XMLHttpRequest.x.onreadystatechange (https://cdn.knightlab.com/libs/timeline3/latest/js/timeline.js:11:117224)

**Comment by JoeGermuska** [comment]
This doesn't seem to be a categorical problem with TimelineJS.

If you can share a URL to a timeline which demonstrates this problem consistently, we can investigate further. If you'd rather not share that on a public GitHub issue, [you can file a support request](https://knightlab.zendesk.com/hc/en-us/requests/new).

**Comment by edgarnegronrabell** [comment]
Apologies. I've been trying to look for answers to solve this issue and I just haven't had any luck. I'm currently working with this in my local environment so I don't have a url for this. May I ask a few questions about the inputs the timeline should receive? I think I've narrowed the problem a bit, but there are some doubts that I haven't been able to clear up with the documentation yet.

**Comment by JoeGermuska** [comment]
Nothing to apologize for; we can try to answer some questions, sure. 

Since your error comes from compiled/minimized code, it's hard to be clear on the source of the error -- consider using the sourcemaps to make that easier -- but it's clear that the error comes as part of some network connection, so it is probably before TimelineJS receives any inputs at all. (It could also be when TimelineJS tries to read a language localization file, if you specify a language other than `en`)

The API via which TimelineJS originally retrieved Google Sheets data has been deprecated for years, and was scheduled to be disabled last fall. Because we have no way of interacting with authors of timelines, we sought a fix for this which would keep old timelines functioning. [This solution is described in some more detail in this blog post.](https://knightlab.northwestern.edu/2020/09/16/timelinejs-update-2/index.html)  Google ended up delaying the disabling of the API once again, but once we launched our changes, it didn't make sense to us to roll them back only temporarily.

**tldr:** TimelineJS can no longer retrieve Google sheets data directly from Google in a browser request. Cross-browser scripting controls require that we go through a simple proxy which gets the data for the browser. The proxy has limited restrictions so that people on the internet with the same problem don't take advantage of our services and bandwidth for non-Knight Lab tools; it's possible that you're getting caught in those restrictions, although the proxy is designed to accept requests for timelines which are not hosted on `*.knightlab.com` domains, as long as the Google Sheets document's header row is exactly the same as the template we offer.

Hope that helps. Ask more questions if you have them. Do consider configuring your development environment to use the sourcemaps, which can be found at 

https://cdn.knightlab.com/libs/timeline3/latest/js/timeline.js.map
https://cdn.knightlab.com/libs/timeline3/latest/css/timeline.css.map

**Comment by edgarnegronrabell** [comment]
This definitely helps a lot. Thank you very much.

**Comment by edgarnegronrabell** [comment]
I just realized I didn't mention that I'm using JSON instead of Google Sheets. I thought at first that it was a format error but it appears to be something else like you said. I will look into it further with the sourcemaps.

**Comment by edgarnegronrabell** [comment]
`GET http://localhost:5000/%7B%22title%22:%7B%22text%22:%7B%22text%22:%22Puerto%20Rico%20Status%20Bills%20Timeline%22%7D%7D,%22events%22:[%7B%22start_date%22:%7B%22year%22:%221900%22,%22month%22:%221%22,%22day%22:%221%22%7D,%22text%22:%7B%22text%22:%22%3Cp%3EA%20Test%3C/p%3E%22%7D%7D,%7B%22start_date%22:%7B%22year%22:%221900%22,%22month%22:%221%22,%22day%22:%221%22%7D,%22text%22:%7B%22text%22:%22%3Cp%3ETo%20Provide%20a%20government%20for%20the%20island%20of%20Puerto%20Rico%20and%20for%20other%20purposes%20(part%20of%20the%20Foraker%20Act%20of%201900)%3C/p%3E%22%7D%7D,%7B%22start_date%22:%7B%22year%22:%222007%22,%22month%22:%222%22,%22day%22:%227%22%7D,%22text%22:%7B%22text%22:%22%3Cp%3ETo%20recognize%20Puerto%20Ricos%20sovereign%20nationhood%20under%20either%20independence%20or%20free%20association%20and%20to%20provide%20for%20a%20transition%20process,%20and%20for%20other%20purposes.%3C/p%3E%22%7D%7D]%7D 404 (NOT FOUND)` 


this is what I get before the TLError, what I've been wondering here is where is it trying to send the GET request, I'm guessing, its not supposed to go to localhost and that's where the problem may be (?)

**Comment by JoeGermuska** [comment]
Yes, it looks as though you are misunderstanding valid values for the second parameter of the `TL.Timeline` constructor.

If the argument passed is a string, it should be a URL which can be used to retrieve configuration data, or, optionally, just the Google Sheets document identifier part. It looks as though you're sending in a JSON String.

If that's correct, then you should instead invoke `JSON.parse(...)` with that string, and pass the actual javascript object as the second argument.  I've made an edit to the constructor documentation in hopes of making this more clear.
